src/components/forms/CreateFurnitureForm.vue

```diff
@@ -1,55 +1,63 @@
 <template>
   <section class="general-form">
-    <form @submit.prevent="generateFurniture">
+    <form @submit.prevent="createFurniture">
       <div>
         <label>Altura en cm?</label>
-        <input type="number" v-model="$store.state.furniture.totalHeight" />
+        <input type="number" v-model="furniture.totalHeight" />
       </div>
       <div>
         <label>Ancho en cm?</label>
-        <input type="number" v-model="$store.state.furniture.totalWidth" />
+        <input type="number" v-model="furniture.totalWidth" />
       </div>
       <div>
         <label>Cuantos espacios horizontales ?</label>
-        <input type="number" v-model="$store.state.furniture.shelfAmount" />
+        <input type="number" v-model="furniture.shelfsAmount" />
       </div>
       <button>Ver mueble!</button>
     </form>
   </section>
 </template>
 
 <script>
-import { mapGetters } from 'vuex';
 export default {
   data() {
-    return {};
+    return {
+      furniture: {
+        shelfs: [],
+        totalHeight: 0,
+        totalWidth: 0,
+        shelfsAmount: 0,
+      },
+    };
   },
   computed: {
-    ...mapGetters(['totalHeight', 'shelfAmount']),
     shelfHeight() {
-      return this.totalHeight / this.shelfAmount;
+      return this.furniture.totalHeight / this.furniture.shelfsAmount;
     },
   },
   methods: {
-    pushShelfData(i) {
+    createFurniture() {
+      this.createShelfs();
+      this.$store.state.furniture = this.furniture;
+    },
+    createShelfs() {
+      this.furniture.shelfs = [];
+      for (let id = 0; id < this.furniture.shelfsAmount; id++) {
+        this.furniture.shelfs.push(this.pushShelf(id));
+      }
+    },
+    pushShelf(id) {
       return {
-        id: i,
+        id: id,
         height: this.shelfHeight,
         confirmed: false,
         insideSeparators: {
           amountOfSeparators: 0,
           typeOfSeparators: 'centered',
-          widthInPercentaje: null,
+          widthOfEachSpace: [],
```

Comment:
with that name, I'm expecting to have a variable that stores a number (somehting related to width).
Instead, I have an array (so a collection of several things).

It is good practice to put it **plural**.

Example: 

```javascript
let houses = [];
some_house = {};
houses.push(some_house);
```

So you immediately understand that it's a collection of houses.

I'm not sure what to expect here. It's a collection of... widths? (of what?) Of spaces?